rand_distr/tests/pdf.rs

```diff
@@ -0,0 +1,86 @@
+// Copyright 2021 Developers of the Rand project.
+//
+// Licensed under the Apache License, Version 2.0 <LICENSE-APACHE or
+// https://www.apache.org/licenses/LICENSE-2.0> or the MIT license
+// <LICENSE-MIT or https://opensource.org/licenses/MIT>, at your
+// option. This file may not be copied, modified, or distributed
+// except according to those terms.
+
+#![allow(clippy::float_cmp)]
+
+use average::Histogram;
+use rand::{Rng, SeedableRng};
+use rand_distr::Normal;
+
+average::define_histogram!(hist, 100);
+use hist::Histogram as Histogram100;
+
+mod sparkline;
+
+#[test]
+fn normal() {
+    const N_SAMPLES: u64 = 1_000_000;
+    const MEAN: f64 = 2.;
+    const STD_DEV: f64 = 0.5;
+    const MIN_X: f64 = -1.;
+    const MAX_X: f64 = 5.;
+
+    let dist = Normal::new(MEAN, STD_DEV).unwrap();
+    let mut hist = Histogram100::with_const_width(MIN_X,MAX_X);
+    let mut rng = rand::rngs::SmallRng::seed_from_u64(1);
+
+    for _ in 0..N_SAMPLES {
+        let _ = hist.add(rng.sample(dist));  // Ignore out-of-range values
+    }
+
+    println!("Sampled normal distribution:\n{}",
+        sparkline::render_u64_as_string(hist.bins()));
+
+    fn pdf(x: f64) -> f64 {
+        (-0.5 * ((x - MEAN) / STD_DEV).powi(2)).exp() /
+            (STD_DEV * (2. * core::f64::consts::PI).sqrt())
+    }
+
+    let mut bin_centers = hist.centers();
+    let mut expected = [0.; 100];
```

Comment:
Unfortunately that does not work, because `hist::LEN` is not public.

I should make it public in a future version of `average`. I can still define a constant for now.